numba_rvsdg/core/datastructures/scfg.py

```diff
@@ -652,16 +652,12 @@ def insert_block_and_control_blocks(
                 # replace previous successor with synth_assign
                 jt[jt.index(s)] = synth_assign
             # finally, replace the jump_targets
-            self.add_block(
-                self.graph.pop(name).replace_jump_targets(
-                    jump_targets=tuple(jt)
-                )
-            )
+            self.graph[name].replace_jump_targets(jump_targets=jt)
         # initialize new block, which will hold the branching table
         new_block = SyntheticHead(
             name=new_name,
-            _jump_targets=tuple(successors),
-            backedges=tuple(),
+            _jump_targets=list(successors),
```

Comment:
```suggestion
            _jump_targets=successors,
```

`successors` is already a list here and doesn't need to be cast into one.